crates/types/src/logs/metadata.rs

```diff
@@ -54,26 +57,123 @@ impl SegmentIndex {
     }
 }
 
-/// Log metadata is the map of logs known to the system with the corresponding chain.
-/// Metadata updates are versioned and atomic.
-#[serde_as]
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone)]
+pub struct LogletRef<P> {
+    pub params: P,
+    pub references: SmallVec<[(LogId, SegmentIndex); 1]>,
```

Comment:
Is it a `SmallVec` to prepare for a future where we are sharing loglets between different logs (e.g. when splitting)?

Reply:
Yes.